index.js

```diff
@@ -40,17 +40,24 @@ function wrapAnsi(code) {
 	return ESCAPES[0] + '[' + code + 'm';
 }
 
-module.exports = function (str, cols) {
-	var pre = '';
-	var ret = '';
+// calculate the length of words split on ' ', ignoring
+// the extra characters added by ansi escape codes.
+function wordLengths(str) {
+	return stripAnsi(str).split(' ').map(function (s) {
+		return s.length;
```

Comment:
Use https://github.com/sindresorhus/string-width here so we handle fullwidth characters correctly.